caom2/caom2/obs_reader_writer.py

```diff
@@ -1467,6 +1494,17 @@ def _add_enity_attributes(self, entity, element):
         if entity._last_modified is not None:
             self._add_attribute(
                 "lastModified", caom_util.date2ivoa(entity._last_modified), element)
+        
+        if self._output_version >= 23:
```

Comment:
That's a good idea but happens to be awkward with the ChecksumURI objects--the attribute value comes from the URI which is a property of of the thing which could be None.  So, I think it's better to keep the None check in this function.